vermouth/processors/stash_attributes.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2025 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Provides a processor that stores attributes 
+"""
+
+from .processor import Processor
+from ..log_helpers import StyleAdapter, get_logger
+
+LOGGER = StyleAdapter(get_logger(__name__))
+
+
+def stash_attributes(molecule, attributes):
+    """
+    For each node in molecule, add the attributes to a stash dictionary
+
+    Parameters
+    ----------
+    molecule: :class:`~vermouth.molecule.Molecule`
+        The molecule to transform.
+    attributes: tuple[str]
+        Attributes to store in the nodes that may otherwise be modified
+    """
+    for attr in attributes:
+        for node in molecule.nodes:
+            # create the stash if not already there
+            if not molecule.nodes[node].get("stash"):
+                molecule.nodes[node]["stash"] = {}
+            # stash the attribute if it hasn't already been. Otherwise raise warning.
+            if not molecule.nodes[node]["stash"].get(attr):
+                molecule.nodes[node]["stash"][attr] = molecule.nodes[node].get(attr, None)
+            else:
+                LOGGER.warning("Trying to stash an already stashed attribute to molecule. Will not stash.")
```

Comment:
```suggestion
            # create the stash if not already there
            stash = molecule.nodes[node].get('stash')
            molecule.nodes[node]["stash"] = stash
            # stash the attribute if it hasn't already been. Otherwise raise warning.
            if attr not in stash:
                stash[attr] = copy.deepcopy(molecule.nodes[node].get(attr, None))
            else:
                LOGGER.warning("Trying to stash already stashed attribute {attr} to molecule. Will not stash.", attr)
```